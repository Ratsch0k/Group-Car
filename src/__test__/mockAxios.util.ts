jest.mock('axios');
import axios from 'axios';
const mockedAxios = axios as jest.Mocked<typeof axios>;

mockedAxios.head.mockImplementationOnce((path: string) => {
  if (path === '/auth') {
    return Promise.resolve({
      headers: {
        'xsrf-token': 'TEST TOKEN',
      },
    });
  }
});

export default mockedAxios;
