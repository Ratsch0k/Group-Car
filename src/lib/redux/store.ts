import {configureStore, getDefaultMiddleware} from '@reduxjs/toolkit';
import rootReducers from './slices';
import {
  routerMiddleware,
} from 'connected-react-router';
import history from 'lib/redux/history';
import * as Sentry from '@sentry/react';

const sentryReduxEnhancer = Sentry.createReduxEnhancer({
  actionTransformer: (action) => {
    return action;
  },
  stateTransformer: (state) => {
    return state;
  },
});


/**
 * Create redux store.
 */
const store = configureStore({
  reducer: rootReducers(history),
  devTools: true,
  middleware: getDefaultMiddleware().concat(routerMiddleware(history)),
  enhancers: [sentryReduxEnhancer],
});

export type RootState = ReturnType<typeof store.getState>;

export type AppDispatch = typeof store.dispatch;

export default store;
