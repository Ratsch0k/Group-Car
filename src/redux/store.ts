import {configureStore} from '@reduxjs/toolkit';
import rootReducer from './slices';

/**
 * Create redux store.
 */
const store = configureStore({
  reducer: rootReducer,
  devTools: true,
});

export type RootState = ReturnType<typeof store.getState>;

export type AppDispatch = typeof store.dispatch;

export default store;
