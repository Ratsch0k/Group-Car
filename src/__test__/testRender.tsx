import React from 'react';
import {render, RenderResult} from '@testing-library/react';
import {ConnectedRouter} from 'connected-react-router';
import {Provider} from 'react-redux';
import history from '../lib/redux/history';
import {StylesProvider, ThemeProvider} from '@material-ui/styles';
import testTheme from './testTheme';
import configureMockStore, {
  MockStoreCreator,
  MockStoreEnhanced,
} from 'redux-mock-store';
import thunk from 'redux-thunk';
import {RootState} from '../lib/redux/store';
import {
  initialState as authState,
} from '../lib/redux/slices/auth/authSlice';
import {
  initialState as groupState,
} from '../lib/redux/slices/group/groupSlice';
import {
  initialState as invitesState,
} from '../lib/redux/slices/invites/invitesSlice';


export interface TestRenderResult<T> {
  mockStore: MockStoreCreator<T>;
  store: MockStoreEnhanced<unknown>;
}

const defaultState: RootState = {
  router: {
    action: 'POP',
    location: {
      pathname: '/',
      search: '',
      hash: '',
      query: {},
      state: undefined,
    },
  },
  auth: authState,
  group: groupState,
  invites: invitesState,
};

/**
 * Testing generateClassName function to
 * avoid weird fails due to random class names
 * @param rule Rule
 * @param stylesheet Stylesheet
 * @returns The class name
 */
const generateClassName = (rule, stylesheet) => {
  return `${stylesheet.options.classNamePrefix}-${rule.key}`;
};


/**
 * Render function for testing purposes.
 * Provides the necessary base components and
 * contexts.
 * @param children The children to render
 * @returns The result of the render function.
 */
function testRender<T>(state: T, children: React.ReactNode):
RenderResult & TestRenderResult<T> {
  const middleware = [thunk];
  const mockStore = configureMockStore<T>(middleware);
  const initialState = {
    ...defaultState,
    ...state,
  };
  const store = mockStore(initialState);

  const renderResult = render(
    <Provider store={store}>
      <ConnectedRouter history={history}>
        <ThemeProvider theme={testTheme}>
          <StylesProvider generateClassName={generateClassName}>
            {children}
          </StylesProvider>
        </ThemeProvider>
      </ConnectedRouter>
    </Provider>,
  );

  return {
    ...renderResult,
    mockStore,
    store,
  };
}

export default testRender;
