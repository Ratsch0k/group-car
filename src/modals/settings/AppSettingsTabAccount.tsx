import React, {FC} from 'react';
import {
  SettingsTabContent,
  SettingsTabContentProps,
} from '../../lib/components/Settings';
import UserAvatar from '../../lib/components/UserAvatar';
import {useShallowAppSelector} from '../../lib/redux/hooks';
import {getUser} from '../../lib/redux/slices/auth';
import {createStyles, makeStyles, Typography} from '@material-ui/core';
import shadows from '@material-ui/core/styles/shadows';
import {useTranslation} from 'react-i18next';
import {GroupCarTheme} from '../../lib';

const useStyles = makeStyles((theme: GroupCarTheme) => createStyles({
  avatar: {
    boxShadow: shadows[5],
  },
  info: {
    marginTop: theme.spacing(1),
  },
}));

export const AppSettingsTabAccount: FC<SettingsTabContentProps<string>> = (
  props,
) => {
  const {index, value} = props;
  const user = useShallowAppSelector(getUser);
  const classes = useStyles();
  const {t} = useTranslation();

  return (
    <SettingsTabContent index={index} value={value}>
      <UserAvatar
        userId={user && user.id}
        size={100}
        className={classes.avatar}
      />
      <Typography align='center'>
        <b>{t('misc.email')}:</b>{' '}
        {user?.email}
      </Typography>
    </SettingsTabContent>
  );
};

export default AppSettingsTabAccount;
