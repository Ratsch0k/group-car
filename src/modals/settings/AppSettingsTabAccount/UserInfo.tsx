import UserAvatar from '../../../lib/components/UserAvatar';
import {
  Box,
  createStyles,
  Grid,
  makeStyles,
} from '@material-ui/core';
import React from 'react';
import {useTranslation} from 'react-i18next';
import {useShallowAppSelector} from '../../../lib/redux/hooks';
import {getUser} from '../../../lib/redux/slices/auth';
import {GroupCarTheme} from '../../../lib';
import AttributeField from '../../../lib/components/AttributeField';
import {
  AttributeFieldProps,
} from '../../../lib/components/AttributeField/AtttributeField';

const useStyles = makeStyles((theme: GroupCarTheme) => createStyles({
  avatar: {
    border: `${theme.palette.divider} 2px solid`,
    margin: '0px !important',
  },
  values: {
    marginTop: theme.spacing(2),
  },
}));

const Field = (props: AttributeFieldProps) => {
  return (
    <Grid item>
      <AttributeField {...props} />
    </Grid>
  );
};

export const UserInfo = (): JSX.Element => {
  const {t} = useTranslation();
  const user = useShallowAppSelector(getUser);
  const classes = useStyles();

  return (
    <>
      <UserAvatar
        userId={user && user.id}
        size={150}
        className={classes.avatar}
      />
      <Box className={classes.values}>
        <Grid container direction='column' wrap='nowrap'>
          <Field label={t('misc.username')}>
            {user?.username}
          </Field>
          <Field label={t('misc.email')}>
            {user?.email}
          </Field>
          <Field label={t('misc.createdAt')}>
            {
              user !== undefined &&
              new Date(user.createdAt).toLocaleString()
            }
          </Field>
        </Grid>
      </Box>
    </>
  );
};
