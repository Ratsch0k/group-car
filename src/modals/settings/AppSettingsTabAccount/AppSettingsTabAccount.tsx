import React, {FC} from 'react';
import {
  SettingsTabContent,
  SettingsTabContentProps,
} from '../../../lib/components/Settings';
import {
  Container,
} from '@material-ui/core';
import {useTranslation} from 'react-i18next';
import {UserInfo} from './UserInfo';
import SettingsSection from '../../../lib/components/SettingsSection';
import Security from './Security';


export const AppSettingsTabAccount: FC<SettingsTabContentProps<string>> = (
  props,
) => {
  const {index, value} = props;
  const {t} = useTranslation();

  return (
    <SettingsTabContent
      index={index}
      value={value}
      id='settings-account-tab-content'
    >
      <Container maxWidth='md'>
        <SettingsSection title={t('settings.account.userInfo')}>
          <UserInfo />
        </SettingsSection>
        <SettingsSection title={t('settings.account.security.title')}>
          <Security />
        </SettingsSection>
      </Container>
    </SettingsTabContent>
  );
};

export default AppSettingsTabAccount;
