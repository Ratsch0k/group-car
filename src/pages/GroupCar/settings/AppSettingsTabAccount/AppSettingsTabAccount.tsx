import React, {FC} from 'react';
import {
  Box,
} from '@material-ui/core';
import {useTranslation} from 'react-i18next';
import {UserInfo} from './UserInfo';
import SettingsSection from 'lib/components/SettingsSection';
import Security from './Security';
import SettingsTabTitle from 'lib/components/Settings/SettingsTabTitle';


export const AppSettingsTabAccount: FC = () => {
  const {t} = useTranslation();

  return (
    <>
      <SettingsTabTitle>
        {t('settings.account.title')}
      </SettingsTabTitle>
      <Box>
        <SettingsSection title={t('settings.account.userInfo')}>
          <UserInfo />
        </SettingsSection>
        <SettingsSection title={t('settings.account.security.title')}>
          <Security />
        </SettingsSection>
      </Box>
    </>
  );
};

export default AppSettingsTabAccount;
