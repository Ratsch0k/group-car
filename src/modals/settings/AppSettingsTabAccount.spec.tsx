import "../../__test__/mockAxios";
import {RootState} from "../../lib/redux/store";
import testRender from "../../__test__/testRender";
import AppSettingsTabAccount from "./AppSettingsTabAccount";

describe('AppSettingsTabAccount', () => {
  let state: Partial<RootState>;

  beforeEach(() => {
    state = {
      auth: {
        user: {
          id: 1,
          username: 'test',
          email: 'test@mail.com',
          createdAt: new Date(10),
          updatedAt: new Date(10),
        }
      }
    }
  });

  it('renders correctly', () => {
    let {baseElement} = testRender(
      state,
      <AppSettingsTabAccount index='/settings/account' value='/settings/account' />,
    );

    expect(baseElement).toMatchSnapshot();
  });
});