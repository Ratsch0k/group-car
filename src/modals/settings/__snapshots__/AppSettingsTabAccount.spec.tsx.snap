// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppSettingsTabAccount renders correctly 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root MuiBox-root makeStyles-root"
      id="settings-account-tab-content"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthMd"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular makeStyles-avatar makeStyles-avatar"
        >
          <img
            alt="user-1"
            class="MuiAvatar-img"
            src="/api/user/1/profile-pic"
          />
        </div>
        <div
          class="MuiContainer-root makeStyles-values MuiContainer-maxWidthSm"
        >
          <div
            class="MuiGrid-root MuiGrid-container"
          >
            <div
              class="MuiGrid-root makeStyles-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item"
            >
              <div
                class="MuiGrid-root makeStyles-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-auto"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 MuiTypography-colorPrimary"
                >
                  <b>
                    misc.username
                  </b>
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-auto"
              >
                test
              </div>
            </div>
            <div
              class="MuiGrid-root makeStyles-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item"
            >
              <div
                class="MuiGrid-root makeStyles-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-auto"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 MuiTypography-colorPrimary"
                >
                  <b>
                    misc.email
                  </b>
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-auto"
              >
                test@mail.com
              </div>
            </div>
            <div
              class="MuiGrid-root makeStyles-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item"
            >
              <div
                class="MuiGrid-root makeStyles-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-auto"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 MuiTypography-colorPrimary"
                >
                  <b>
                    settings.account.createdAt
                  </b>
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-auto"
              >
                1/1/1970, 1:00:00 AM
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
